src/convertToObject.js

```diff
@@ -6,7 +6,28 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const cleanedString = sourceString.trim();
+
+  const declarations = cleanedString.split(';');
+
+  const result = {};
+
+  for (const declaration of declarations) {
+    const trimmedDeclaration = declaration.trim();
+
+    if (trimmedDeclaration) {
+      const parts = trimmedDeclaration.split(':');
+
+      if (parts.length === 2) {
+        const property = parts[0].trim();
+        const value = parts[1].trim();
+
+        result[property] = value;
+      }
+    }
+  }
```

Comment:
[CODE PRACTICE] - Avoid using loops in this task, use iteration methods instead. You can replace the `for...of` loop with `forEach`, which is an exception to the rule against modifying existing arrays/objects with iteration methods.